auctioneer/client.go

```diff
@@ -892,7 +897,11 @@ func (c *Client) connectServerStream(initialBackoff time.Duration,
 	// Now that we know the connection itself is established, we also re-
 	// connect the long-lived stream.
 	ctx, c.streamCancel = context.WithCancel(ctxb)
-	c.serverStream, err = c.client.SubscribeBatchAuction(ctx)
+	if c.cfg.ConnectSidecar {
+		c.serverStream, err = c.client.SubscribeSidecar(ctx)
```

Comment:
No, it should work directly since the sidecar connection client is a secondary instance with its own connection. This is an interesting test case for an integration test as well :+1: